### PR TITLE
fix(Core/Totem): Implement casting delay for the Fire Totem used by c…

### PR DESCRIPTION
**Pull request by Nyeriah**
…reatures

## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14262
- Closes https://github.com/chromiecraft/chromiecraft/issues/4579

## Tests Performed:

- Tested ingame
- 

## How to Test the Changes:

1. Cast 32062
2. fire nova happens after 4s instead of on spawn
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Entities/Totem/Totem.h [hunk 1]
```diff
@@ -27,7 +27,12 @@ enum TotemType
     TOTEM_STATUE     = 2 // copied straight from moongose, may need more implementation to work
 };
 // Some Totems cast spells that are not in creature DB
-#define SENTRY_TOTEM_SPELLID  6495
+
+enum TotemSpellIds : uint32
+{
+    SentryTotemSpell = 6495,
```
Comment (Nyeriah) [inline src/server/game/Entities/Totem/Totem.h:33]: @Kitzunu is the case here correct? I don't remember what format is used for this kind of enum

Comment (Kitzunu) [reply src/server/game/Entities/Totem/Totem.h:33]: it works, but using enum class would be safer

Comment (Kitzunu) [reply src/server/game/Entities/Totem/Totem.h:33]: enum class TotemSpellIds: uint32

Comment (Kitzunu) [reply src/server/game/Entities/Totem/Totem.h:33]: https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

File: src/server/game/Entities/Totem/Totem.h [hunk 2]
```diff
@@ -27,7 +27,12 @@ enum TotemType
     TOTEM_STATUE     = 2 // copied straight from moongose, may need more implementation to work
 };
 // Some Totems cast spells that are not in creature DB
-#define SENTRY_TOTEM_SPELLID  6495
+
+enum TotemSpellIds : uint32
```
Comment (Nyeriah) [inline src/server/game/Entities/Totem/Totem.h:31]:
```suggestion
enum class TotemSpellIds : uint32
```

**Comment by Nyeriah** [comment]
@Kitzunu please review when you can, I think it`s working now

File: src/server/game/Entities/Totem/Totem.cpp [hunk 3]
```diff
@@ -132,7 +146,7 @@ void Totem::UnSummon(uint32 msTime)
 
         // Remove Sentry Totem Aura
         if (GetEntry() == SENTRY_TOTEM_ENTRY)
-            owner->RemoveAurasDueToSpell(SENTRY_TOTEM_SPELLID);
+            owner->RemoveAurasDueToSpell((uint32)TotemSpellIds::SentryTotemSpell);
```
Comment (Kitzunu) [inline src/server/game/Entities/Totem/Totem.cpp:149]: preferably a static_cast otherwise it looks good

Comment (Nyeriah) [reply src/server/game/Entities/Totem/Totem.cpp]: Done, ty

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**